src/app/features/reporting/query.service.spec.ts

```diff
@@ -455,25 +464,94 @@ describe("QueryService", () => {
 
   it("should do addPrefix as part of toEntities if optional parameter is given", async () => {
     const queryWithAddPrefix = `
-      ${School.ENTITY_TYPE}:toArray[*privateSchool=true]
+      ${School.ENTITY_TYPE}:toArray
       :getRelated(${ChildSchoolRelation.ENTITY_TYPE}, schoolId)
       .childId:addPrefix(${Child.ENTITY_TYPE}):toEntities.name`;
     const queryWithoutAddPrefix = `
-      ${School.ENTITY_TYPE}:toArray[*privateSchool=true]
+      ${School.ENTITY_TYPE}:toArray
       :getRelated(${ChildSchoolRelation.ENTITY_TYPE}, schoolId)
       .childId:toEntities(${Child.ENTITY_TYPE}).name`;
 
-    const resultWithAddPrefix = await service.queryData(
-      queryWithAddPrefix,
-      null,
-      null
-    );
+    const resultWithAddPrefix = await service.queryData(queryWithAddPrefix);
     const resultWithoutAddPrefix = await service.queryData(
-      queryWithoutAddPrefix,
-      null,
-      null
+      queryWithoutAddPrefix
     );
 
+    expect(resultWithoutAddPrefix).toHaveSize(3);
     expect(resultWithoutAddPrefix).toEqual(resultWithAddPrefix);
   });
+
+  it("should create an attendance array with the current school", async () => {
+    const attendanceArrayQuery = `${EventNote.ENTITY_TYPE}:toArray:getAttendanceArray(true)`;
+
+    const attendanceResult: AttendanceInfo = await service.queryData(
+      attendanceArrayQuery
+    );
+
+    expect(attendanceResult).toContain({
+      participant: "maleChild",
+      school: "privateSchool",
+      status: new EventAttendance(presentAttendanceStatus),
+    });
+    expect(attendanceResult).toContain({
+      participant: "femaleChristianChild",
+      status: new EventAttendance(presentAttendanceStatus),
+    });
+    expect(attendanceResult).toContain({
+      participant: "maleChild",
+      school: "privateSchool",
+      status: new EventAttendance(absentAttendanceStatus),
+    });
+    expect(attendanceResult).toContain({
+      participant: "femaleMuslimChild",
+      status: new EventAttendance(presentAttendanceStatus),
+    });
+    expect(attendanceResult).toContain({
+      participant: "femaleChristianChild",
+      status: new EventAttendance(presentAttendanceStatus),
+    });
+    expect(attendanceResult).toContain({
+      participant: "maleChild",
+      status: new EventAttendance(absentAttendanceStatus),
+    });
+    expect(attendanceResult).toContain({
+      participant: "femaleChristianChild",
+      status: new EventAttendance(absentAttendanceStatus),
+    });
+    expect(attendanceResult).toContain({
+      participant: "maleChristianChild",
+      status: new EventAttendance(presentAttendanceStatus),
+    });
+  });
+
+  it("should create a attendance report with percentages", async () => {
+    const reportQuery = `${EventNote.ENTITY_TYPE}:toArray:getAttendanceArray:getAttendanceReport`;
```

Comment:
is there any feasible way to set up the data here in the unit test itself? with a 500+ lines test file it becomes hard to match what the "expect()" result should be